File: presentation/schemas/configuration.ts
export interface UserConfigurationSchema {
    root: boolean;
    admin: boolean;
    user: boolean;
    soport: boolean;
}

export interface NotificationConfigurationSchema {
    ifName: boolean;
    ifDescr: boolean;
    ifAlias: boolean;
    ifHighSpeed: boolean;
    ifOperStatus: boolean;
    ifAdminStatus: boolean;
}

export interface ConfigurationSchema {
    can_assign: UserConfigurationSchema;
    can_receive_assignment: UserConfigurationSchema;
    notification_changes: NotificationConfigurationSchema;
    view_information_global: UserConfigurationSchema;
}

File: icm/business/models/configuration.py
from pydantic import BaseModel


class ConfigUser(BaseModel):
    root: bool
    admin: bool
    user: bool
    soport: bool


class ConfigInterface(BaseModel):
    ifName: bool
    ifDescr: bool
    ifAlias: bool
    ifHighSpeed: bool
    ifOperStatus: bool
    ifAdminStatus: bool


class NewConfigModel(BaseModel):
    can_assign: ConfigUser
    can_receive_assignment: ConfigUser
    view_information_global: ConfigUser
    notification_changes: ConfigInterface
